chajda/tsquery/augments.py

```diff
@@ -53,114 +58,225 @@ def augments_gensim(lang, word, config=Config(), n=5):
+    fasttext_models[lang].get_nearest_neighbors('google', k=5)
+    fasttext_list = []
+    for word in fasttext_models[lang].get_words(on_unicode_error='replace'):
+        fasttext_list.append(word)
+
+
+def destroy_fasttext(lang):
+    '''
+    removes fasttext model from disk and memory
+    '''
+
+    # remove fasttext download from disk
+    if os.path.isfile(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin"):
+        os.remove(f"{fasttext_annoy_data_path}/fasttext/cc.{lang}.300.bin")
+
+    # remove fasttext model from memory
+    try:
+        del fasttext_models[lang]
+    except KeyError:
+        pass
+
+
+#################################################################################
+# annoy
+#################################################################################
+
+# annoy_indices dictionary will contain languages as keys,
+# and a namedtuple containing the corresponding annoy index, as well as two json files, as values
 annoy_indices = {}
 
+# fasttext_index_for_word dictionary will contain keys for all words in a fasttext language dictionary,
+# and values corresponding to the index of a word;
+# it is populated by the load_annoy_index function,
+# and subsequently written to a json file
+# NOTE:
+# this dictionary is necessary for using annoy for nearest neighbor query after removing fasttext download
+fasttext_index_for_word = {}
 
-def create_annoy_index_if_needed(lang, word='google', n=5, seed=22):
+# fasttext_word_at_index dictionary will contain values for all words in a fasttext language dictionary,
+# and keys corresponding to the index of a word;
+# it is populated by the load_annoy_index function,
+# and subsequently written to a json file
+# NOTE:
+# this dictionary is necessary for using annoy for nearest neighbor query after removing fasttext download
+fasttext_word_at_index = {}
+
+
+def load_annoy(lang, word='google', n=5, seed=22, run_destroy_fasttext = True):
     '''
     Checks whether an annoy index has already been created for a specified language.
     Populates an annoy index  with vectors from the fasttext model that corresponds to language,
-    builds the index, saves it to disk. 
-    The annoy index is stored in the annoy_indices dictionary.
-
+    builds the index, saves it to disk.
+    The annoy index is stored in the annoy dictionary.
     lang: language passed from augments_fasttext
-    
+
     NOTE:
     Populating an annoy index with vectors from a fasttext model requires more memory than github actions allows for.
     Therefore, the doctest has been commented out.
-
     # >>> create_annoy_index_if_needed('en') or annoy_indices['en'] != None
     # True
     '''
     # create annoy index if not already created
     try:
         annoy_indices[lang]
     except KeyError:
-        annoy_indices[lang] = AnnoyIndex(300, 'angular')
+        annoy_indices[lang] = namedtuple(f"annoy", ['index', 'json1', 'json2'])
```

Comment:
`json1`/`json2` are not good names for these variables, especially since you have different names for the global dictionaries they represent